sys/include/net/nanocoap.h

```diff
@@ -762,6 +799,91 @@ static inline ssize_t coap_get_location_query(const coap_pkt_t *pkt,
                                target, max_len, '&');
 }
 
+/**
+ * @brief Initialize a block2 response
+ *
+ * This function initializes a block2 response and adds the block2 option to
+ * the packet
+ *
+ * Caller must ensure that this function is called after all required
+ * preceding options are added
+ *
+ * @param[in]   buf         buffer to write to
+ * @param[in]   lastonum    last option number (must be <27)
+ * @param[in]   pkt         packet to work on
+ * @param[out]  blk         Preallocated blockbuilder struct to fill
+ *
+ * @returns     amount of bytes written to @p buf
+ */
+size_t coap_block2_init(uint8_t* buf, uint16_t lastonum, coap_pkt_t *pkt, coap_blockbuilder_t *blk);
```

Comment:
Let me try something here, I think I understand what you want and it makes sense to me.

Reply:
@kb2ma I've reworked it to use only a `coap_block_slicer_t` here. It required 2 helper function which added 20 bytes of additional flash compared to previous fixup state. This prevents us from having to put that `coap_block1_t` struct on the stack during the full handler processing.

Please let me know if this is workable for gcoap since I didn't completely follow your instructions :)